From 61fd7d6720c562c88ab79062ff8d131e5e3c7b1b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?L=C3=A9na=C3=AFc=20Huard?= <lenaic@lhuard.fr>
Date: Wed, 1 Apr 2020 23:45:08 +0200
Subject: [PATCH] udev: Fix SIGSEGV in AlternativeNamesPolicy handling

The function sd_device_get_property_value has some paths where it exits without
touching the n pointer. In those cases, n remained uninitialized until it was
eventually read inside isempty where it caused the segmentation fault.

Fixes #15078
---
 src/udev/net/link-config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev/net/link-config.c b/src/udev/net/link-config.c
index 71946c9ecc75..48039511fa69 100644
--- a/src/udev/net/link-config.c
+++ b/src/udev/net/link-config.c
@@ -498,7 +498,7 @@ int link_config_apply(link_config_ctx *ctx, link_config *config,
 
         if (config->alternative_names_policy)
                 for (NamePolicy *p = config->alternative_names_policy; *p != _NAMEPOLICY_INVALID; p++) {
-                        const char *n;
+                        const char *n = NULL;
 
                         switch (*p) {
                         case NAMEPOLICY_DATABASE:
